### ISSUE TITLE
Changing `offset` instead of `start_time` when setting `time` in DADA (and GUPPI) headers

### ISSUE THREAD

**Issue by cczhu** [mhvk/baseband#224]
DADA (and GUPPI; #212) headers have both `start_time` and `offset` values, allowing the user to specify the current and start times independently.  Right now, setting `time` will alter `start_time` and keep `offset` fixed.  It might make more sense to do the reverse.

The trouble with setting `offset` rather than `start_time` is that for most other formats we only define a `time` when writing.  Since `offset` is by default zero, passing `time` to a DADA or GUPPI writer properly sets the start time, but setting `offset` in the setter for `time` means we'd need something like:

```
if 'STT_IMJD' not in self.keys():
    self.offset = 0
    self.start_time = time
else:
    self.offset = time - self.start_time
```

For DADA we'd replace `STT_IMJD` with `UTC_START`.

**Comment by cczhu** [comment]
I don't think we've decided whether to implement the above; @mhvk, what do you think?

**Comment by cczhu** [comment]
As discussed today, we decided to implement the setting in the `time` property.  if the user initialises a header using only `time`, we first check if 'STT_IMJD' is already set; if it isn't, we set `time` to its value, and `start_time` using `offset` (which by default is zero).

**Comment by cczhu** [comment]
Solved by #240
